Kernel/API/ProcessExposed.h

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2021, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Function.h>
+#include <AK/RefCounted.h>
+#include <AK/RefPtr.h>
+#include <AK/String.h>
+#include <AK/Types.h>
+#include <Kernel/Arch/x86/CPU.h>
+#include <Kernel/FileSystem/File.h>
+#include <Kernel/FileSystem/FileSystem.h>
+#include <Kernel/KBufferBuilder.h>
+#include <Kernel/KResult.h>
+#include <Kernel/Process.h>
+#include <Kernel/UserOrKernelBuffer.h>
+
+namespace Kernel {
+
+class ProcFS;
+class ProcFSExposedComponent;
+class ProcFSExposedFolder;
+class ProcFSRootFolder;
+class ProcFSBusDirectory;
+class ProcFSSystemBoolean;
+class ProcFSComponentsRegistrar {
+    friend class ProcFS;
+    friend class ProcFSExposedComponent;
+    friend class ProcFSExposedFolder;
+    friend class ProcFSRootFolder;
+
+public:
+    static ProcFSComponentsRegistrar& the();
+
+    static void initialize();
+
+    InodeIndex allocate_inode_index() const;
+
+    ProcFSComponentsRegistrar();
+    void register_new_bus_folder(ProcFSExposedFolder&);
+
+    NonnullRefPtr<ProcFSBusDirectory> buses_folder() const;
+
+    void register_new_component(ProcFSExposedComponent&);
+    void register_new_process(Process&);
+    void unregister_process(Process&);
+
+    NonnullRefPtr<ProcFSRootFolder> root_folder() { return m_root_folder; }
```

Comment:
Since this function does not pass ownership of the `ProcFSRootFolder` to the caller, there's no need to return a smart pointer. Instead we can return a `ProcFSRootFolder&` and let the caller decide if they want to take ownership (by creating a RefPtr or NonnullRefPtr.)

(This same comment repeated for every instance of this.)